version: '2.3'
services:
  database:
    image: quay.io/azavea/postgis:2.2-postgres9.5-slim
    environment:
      - POSTGRES_USER=planit
      - POSTGRES_PASSWORD=planit
      - POSTGRES_DB=planit
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "planit"]
      interval: 3s
      timeout: 3s
      retries: 3
      start_period: 5s

  angular:
    image: planit-spa
    build:
      context: ./src/angular
      dockerfile: Dockerfile
    volumes:
      - ./src/angular/planit/src:/opt/planit/angular/planit/src
    ports:
      - "4210:4210"
  nginx:
    build:
      context: ./nginx
      dockerfile: Dockerfile
    depends_on:
      - django
    ports: 
      - "8000:443"
    volumes:
      - "./nginx/dist:/var/www"
      - "./nginx/static:/var/static"
      - "./nginx/etc/nginx/conf.d:/etc/nginx/conf.d"
      - "./nginx/etc/nginx/includes/security-headers.conf:/etc/nginx/includes/security-headers.conf"
      - "./nginx/etc/nginx/nginx.conf:/etc/nginx/nginx.conf"
  django:
    image: planit-app
    build:
      context: ./src/django
      dockerfile: Dockerfile
    depends_on:
      database:
        condition: service_healthy
    environment:
      - POSTGRES_HOST=database.service.planit.internal
      - POSTGRES_PORT=5432
      - POSTGRES_USER=planit
      - POSTGRES_PASSWORD=planit
      - POSTGRES_DB=planit
      - DJANGO_ENV=Development
      - DJANGO_DEFAULT_FROM_EMAIL=planit@azavea.com
      - DJANGO_SECRET_KEY=secret
      - DJANGO_LOG_LEVEL=INFO
      - AWS_PROFILE=planit
    command:
      - "--workers=2"
      - "--timeout=60"
      - "--bind=0.0.0.0:8080"
      - "--reload"
      - "--log-level=debug"
      - "--access-logfile=-"
      - "--error-logfile=-"
      - "-kgevent"
      - "planit.wsgi"
    ports:
      - "8100:8080"
      - "8101:8081"
    links:
      - database:database.service.planit.internal
    volumes:
      - ./src/django:/usr/src
      - ./nginx/static:/static
      - $HOME/.aws:/root/.aws:ro
